hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/TestApacheClientConnectionPool.java

```diff
@@ -64,12 +95,14 @@ private void validatePoolSize(int size) throws Exception {
         connections[i] = Mockito.mock(HttpClientConnection.class);
       }
 
-      for (int i = 0; i < size * 2; i++) {
-        keepAliveCache.put(connections[i]);
+      for (int i = 0; i < size; i++) {
+        Assertions.assertThat(keepAliveCache.put(connections[i])).isTrue();
```

Comment:
actually, you could have an assertPut(keepAliveCache, connection) which does this and includes the assert...and use it everywhere

Reply:
Very good point. Have taken it.